### PR TITLE
dev/financial#68 Ensure that check number is correctly passed through…

### PR DESCRIPTION
**Pull request by seamuslee001**
… when processing a contribution or membership batch

Overview
----------------------------------------
This is an alternate to #15264 as it alters the post processing on the batch rather than altering field names as the field_name should be contribution_check_number now

Before
----------------------------------------
Check number is only recorded against the contribution not the payment as well when submitting a contribution or membership batch

After
----------------------------------------
Check number is recorded against the contribution and the payment

ping @monishdeb @eileenmcnaughton

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
Approach makes sense - will let @monishdeb chime in

**Comment by monishdeb** [comment]
@eileenmcnaughton @seamuslee001 I am happy with the patch too. Earlier I was thinking, instead of tackling in the form or BAO level we should remove the deprecated field names from the source and so I changed the field name, and let the BAO fn process it accordingly. On the other hand, if we create a profile with contribution field - check number it again renders the field name as 'contribution_check_number' and thus it will again cause the same issue after form submission until and unless we change the field name from the source. OR let BAO/API to support the old fieldname too.

**Comment by eileenmcnaughton** [comment]
The api should support unique names & real field names at the wrapper level already - but this is bypassing the api

**Comment by seamuslee001** [comment]
Jenkins re test this please

**Comment by seamuslee001** [comment]
@monishdeb @eileenmcnaughton i just added a commit which alters a current contribution api test so that it submits using the unique field name contribution_check_number and then checks that check_number is still correctly stored in the financial trxn records as well as in civicrm_contribution

File: tests/phpunit/api/v3/ContributionTest.php [hunk 1]
```diff
@@ -233,7 +233,7 @@ public function testGetTestContribution() {
    */
   public function testGetContributionReturnFunctionality() {
     $params = $this->_params;
-    $params['check_number'] = 'bouncer';
+    $params['contribution_check_number'] = 'bouncer';
```
Comment (eileenmcnaughton) [inline tests/phpunit/api/v3/ContributionTest.php]: ideally both  should work - maybe  test both?

Comment (seamuslee001) [reply tests/phpunit/api/v3/ContributionTest.php]: Done @eileenmcnaughton

**Comment by eileenmcnaughton** [comment]
Ok - @monishdeb agreed earlier so MOP

**Pull request merged**